gpii/node_modules/WindowsUtilities/WindowsUtilities.js

```
@@ -1060,22 +1111,65 @@ windows.createEmptyStructure = function (structName) {
+windows.combineFlags = function (flagNamesArray, allFlags, op) {
+    allFlags = allFlags || windows.flagConstants;
+    var or = op !== "and" && op !== "&";
+
+    var combine = or
+        ? function (left, right) { return left | right; }
+        : function (left, right) { return left & right; };
+
+    var combinedFlags = or
+        ? 0 : 1;
+
     if (!fluid.isArrayable(flagNamesArray)) {
-        console.log("GPII Windows SpiSettingsHandler combineFlags: array expected!");
+        fluid.log("GPII Windows SpiSettingsHandler combineFlags: array expected!");
         return 0;
     }
     for (var index in flagNamesArray) {
-        combinedFlags = combinedFlags | windows.flagConstants[flagNamesArray[index]];
+        combinedFlags = combine(combinedFlags, allFlags[flagNamesArray[index]]);
     }
     return combinedFlags;
 };
 
+/**
+ * Resolves a list of flags into the numeric value. Flags can either be a string, with the delimiter of either
+ * "|" or "&" specifying the operation, or an array of flag strings (in which case the operation is binary OR).
+ *
+ * The value of each possible flag is specified in allFlags.
+ *
+ * @param {String|Array} flags The list of flags to resolve.
+ * @param {Object} allFlags An associative object of every flag.
+ * @return {Number} The numeric value of flags.
+ */
+windows.resolveFlags = function (flags, allFlags) {
+    var togo = 0;
+    if (!isNaN(Number(flags))) {
+        // Use the hard-coded number as-is
+        togo = Number(flags);
+    } else {
+        var op = "or";
+        // Split a string into an array.
+        if (typeof(flags) === "string") {
+            var or = flags.indexOf("&") < 0;
+            op = or ? "or" : "and";
+            flags = flags.split(/ *[&|,]+ */);
+        }
+        if (fluid.isArrayable(flags)) {
+            togo = gpii.windows.combineFlags(flags, allFlags, op);
+        } else {
+            fluid.log("windows.resolveFlags was passed an unknown type of flags");
```

Fail rather than logging, the community will thank you in the end : P

---

Yes, you are right.